filecheck/src/FileCheck/Command.hs

```diff
@@ -0,0 +1,114 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module FileCheck.Command
+  ( Command(..)
+  , parse
+  , Match(..)
+  , match
+  ) where
+
+import Data.Text qualified as Text
+import Data.Text (Text)
+import FileCheck.Directive (Directive, Prefix)
+import FileCheck.Directive qualified as FCD
+import FileCheck.Pos qualified as FCP
+import FileCheck.Pos (Loc, Span)
+import Prelude hiding (span)
+
+-- | A t'Command' is a pair of a 'Directive' and its argument.
+--
+-- It represents a single line written by the user, e.g., @CHECK: foo@ is parsed
+-- to @v'Command' 'FCD.Check' "foo"@.
```

Comment:
This `Command` value in the Haddocks has two fields, but the actual `Command` data constructor has three fields.